common/src/main/kotlin/com/hartwig/actin/trial/input/FunctionInputResolver.kt

```diff
@@ -981,6 +1007,14 @@ class FunctionInputResolver(
         return gene
     }
 
+    private fun secondParameterAsGene(function: EligibilityFunction): String {
+        val gene = parameterAsString(function, 1)
+        if (!molecularInputChecker.isGene(gene)) {
+            throw IllegalStateException("Not a valid gene: $gene")
+        }
+        return gene
```

Comment:
This looks a bit odd to me... why not have `parameterAsGene` and then pass the parameterAsString as input?